### PR TITLE
search: activate smart search by default

### PR DESCRIPTION
**Pull request by rvantonder**
This PR merges #44395 again after fixing tests. Just stamp because #44395 was already approved and needs to go in for branch cut.

## Test plan

Same as #44395

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

File: schema/settings.schema.json [hunk 1]
```diff
@@ -681,6 +681,11 @@
       "minimum": 0,
       "default": 1
     },
+    "search.defaultMode": {
+      "description": "Defines default properties for search behavior. The default is `smart`, which provides query assistance that automatically runs alternative queries when appropriate. When `precise`, search behavior strictly searches for the precise meaning of the query.",
+      "type": "string",
+      "pattern": "precise|smart"
```
Comment (eseliger) [inline schema/settings.schema.json:687]: if you use `enum: [precise, smart]` instead I think the autocompletion will be more meaningful, but just a drive-by comment - feel free to ignore me :D

**Review by camdencheek — APPROVED** [review]

**Comment by limitedmage** [comment]
Should we update the changelog for this?

File: client/search-ui/src/input/toggles/SmartSearchToggle.tsx [hunk 2]
```diff
@@ -55,6 +55,7 @@ export const SmartSearchToggle: React.FunctionComponent<SmartSearchToggleProps>
                 <PopoverTrigger
                     as={Button}
                     className={classNames(
+                        'a11y-ignore',
```
Comment (limitedmage) [inline client/search-ui/src/input/toggles/SmartSearchToggle.tsx:58]: FYI this is on my radar to fix this week.

**Comment by rvantonder** [comment]
yes, changelog + initial docs will happen, will get something stamped by tomorrow :-)

File: client/search-ui/src/input/toggles/SmartSearchToggle.tsx (hunk 2 above)
Comment (rvantonder) [reply client/search-ui/src/input/toggles/SmartSearchToggle.tsx:58]: np, sounds good!

**Pull request merged**